packages/cozy-client/src/ClientErrors.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useState, useEffect } from 'react'
+import { useClient } from 'cozy-client'
+import QuotaAlert from 'cozy-ui/react/QuotaAlert'
+import { pull } from 'lodash'
+
+/**
+ * Rendering functions for client fetch errors by status code
+ *
+ * @private
+ * @type {object}
+ */
+const byHttpStatus = {
+  413: QuotaError
+}
+
+/**
+ * Display for a quota error from the client
+ *
+ * @see QuotaAlert
+ * @private
+ * @param {Function} dismiss - remove the error from the stack to display
+ * @returns {Function} React rendering
+ */
+function QuotaError({ dismiss }) {
+  return <QuotaAlert onClose={dismiss} />
+}
+
+/**
+ * Renders a stack of errors
+ *
+ * @private
+ * @see ClientErrors
+ * @param {Error[]} errorStack - array of errors/exceptions
+ * @param {Function} setErrorStack - mutates the array of errors
+ * @returns {Function} React rendering
+ */
+function renderErrors(errorStack, setErrorStack) {
+  return errorStack.map((error, key) => {
+    if (error instanceof Response) {
+      const Component = byHttpStatus[error.status]
+      const dismiss = () => setErrorStack(stack => pull(stack, error))
+      return <Component key={key} error={error} dismiss={dismiss} />
+    } else {
+      return null
+    }
+  })
+}
+
+/**
+ * Manage the client errors and exceptions, display them if necessary
+ *
+ * Only Quota Errors are managed for now.
+ *
+ * @param {object} props -
+ * @param {boolean} props.handleExceptions - should cozy-client directly handle errors before forwarding them to the caller?
+ * @param {Function} props.children - React rendering children
+ * @returns {Function} React rendering
+ */
+export function ClientErrors(props) {
+  const { handleExceptions = true, children } = props
+
+  const client = useClient()
+  const [errorStack, setErrorStack] = useState([])
+
+  /**
+   * Handle client errors, add them to the error stack
+   *
+   * @param {Error} error -
+   * @returns {boolean}
+   */
+  const handleError = useCallback(
+    error => {
+      // `isErrorManaged` is there to avoid the same error to be added twice
+      // once the error has been added once, the `isErrorManaged`is set to true
+      // and any future push is ignored
+      if (!error.isErrorManaged) {
+        if (error instanceof Response) {
+          if (byHttpStatus[error.status]) {
+            error.isErrorManaged = true
```

Comment:
The error instanceof Response and byHttpStatus[error.status] could be refactored into a `getErrorIdentifier` so that support for other errors can be added easily.

I am wondering if the getComponentFromError and getIdentifier could be merged into a single function that returns { identifier, component }. This way, no drift can appear between the two functions.